hazelcast/src/main/java/com/hazelcast/internal/networking/nio/NioNetworking.java

```diff
@@ -128,6 +128,35 @@ public IOBalancer getIOBalancer() {
         return ioBalancer;
     }
 
+    @Override
+    public void probeIn(ProbingCycle cycle) {
+        Tags tags = cycle.openContext().tag(TAG_TYPE, "inputThread");
+        for (int i = 0; i < inputThreads.length; i++) {
+            NioThread thread = inputThreads[i];
+            if (thread.isAlive()) {
+                tags.tag(TAG_INSTANCE, thread.getName());
+                cycle.probe("tcp", thread);
+            }
+        }
+        tags = cycle.openContext().tag(TAG_TYPE, "outputThread");
```

Comment:
will this cause litter? a new Tags object for every call to openContext().tag?

Reply:
No. We use this to avoid litter. The context is a `StringBuilder` we write to all the time. As long as the arguments do not create litter before they are passed this is garbage free.

Reply:
ok great.